src/Compilers/CSharp/Portable/Syntax/RecordDeclarationSyntax.cs

```diff
@@ -13,5 +15,22 @@ public partial class RecordDeclarationSyntax
                 return BaseList?.Types.FirstOrDefault() as PrimaryConstructorBaseTypeSyntax;
             }
         }
+
+        public RecordDeclarationSyntax Update(SyntaxList<AttributeListSyntax> attributeLists, SyntaxTokenList modifiers, SyntaxToken keyword, SyntaxToken identifier,
+            TypeParameterListSyntax? typeParameterList, ParameterListSyntax? parameterList, BaseListSyntax? baseList, SyntaxList<TypeParameterConstraintClauseSyntax> constraintClauses,
+            SyntaxToken openBraceToken, SyntaxList<MemberDeclarationSyntax> members, SyntaxToken closeBraceToken, SyntaxToken semicolonToken)
+        {
+            return Update(attributeLists, modifiers, keyword, this.RecordModifier, identifier,
+                typeParameterList, parameterList, baseList, constraintClauses, openBraceToken, members, closeBraceToken, semicolonToken);
+        }
+    }
+}
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    public partial class SyntaxFactory
```

Comment:
>SyntaxFactory [](start = 25, length = 13)

It doesn't feel appropriate to place this type declaration in this file. I think historically we were putting back compat overloads in SytaxFactory.cs.

Reply:
You're right that `SyntaxFactory.cs` does contain a number of such compat APIs.
But we've added syntax and SyntaxFactory compat APIs in separate files for a while now (that's what I've mostly seen done). See `BlockSyntax.cs`, `ForEachVariableStatementSyntax.cs`, `ConstructorDeclarationSyntax.cs`, ...

---
In reply to: [577834483](https://github.com/dotnet/roslyn/pull/51252#discussion_r577834483) [](ancestors = 577834483)